compiler/rustc_middle/src/ty/query/on_disk_cache.rs

```diff
@@ -525,7 +525,7 @@ impl<'sess> OnDiskCache<'sess> {
     ) {
         let mut current_diagnostics = self.current_diagnostics.borrow_mut();
 
-        let x = current_diagnostics.entry(dep_node_index).or_insert(Vec::new());
+        let x = current_diagnostics.entry(dep_node_index).or_insert_with(Vec::new);
```

Comment:
```suggestion
        let x = current_diagnostics.entry(dep_node_index).or_default();
```

Reply:
Applied this one and others, thanks, looks more clear.